### ISSUE TITLE
unsubscribe with limit does not work

### ISSUE THREAD

**Issue by Koseng** [nats-io/nats.py#251]
nats-py version: 2.0.0rc5
nats-server version: 2.6.6

When executing the Hello World example, then "hello" "world" is not printed out after await sub.unsubscribe(limit=2).

```
    # Stop receiving after 2 messages.
    await sub.unsubscribe(limit=2)
    await nc.publish("foo", b'Hello')
    await nc.publish("foo", b'World')
    await nc.publish("foo", b'!!!!!')
```

**Comment by KMilhan** [comment]
https://github.com/nats-io/nats.py/pull/245 fix is merged to main

**Comment by KMilhan** [comment]
for now, you can either wait for the release or skip using `limit` arg (for a while)

**Comment by Koseng** [comment]
Thanks for the info.

**Comment by wallyqs** [comment]
Fixed via #245
